uefi/src/proto/network/pxe.rs

```diff
@@ -432,20 +464,39 @@ impl BaseCode {
 
         let mut buffer_size = buffer.len();
 
+        let mut dest_ip_efi = dest_ip.as_ref().map(|ip| EfiIpAddr::from(**ip));
+        let mut src_ip_efi = src_ip.as_ref().map(|ip| EfiIpAddr::from(**ip));
+
         let status = unsafe {
             (self.0.udp_read)(
                 &mut self.0,
                 op_flags,
-                opt_ip_addr_to_ptr_mut(dest_ip),
+                opt_ip_addr_to_ptr_mut(dest_ip_efi.as_mut()),
                 opt_mut_to_ptr(dest_port),
-                opt_ip_addr_to_ptr_mut(src_ip),
+                opt_ip_addr_to_ptr_mut(src_ip_efi.as_mut()),
                 opt_mut_to_ptr(src_port),
                 header_size,
                 header_ptr,
                 &mut buffer_size,
                 buffer.as_mut_ptr().cast(),
             )
         };
+
+        // Translate IP types back into the higher-level types.
+        let fn_replace_ip = |core_ip: &mut Option<&mut IpAddr>, efi_ip: Option<EfiIpAddr>| {
+            // TODO use let-chain once we are on MSRV 1.88
+            if let Some(core_ip_location) = core_ip {
+                if let Some(efi_ip) = efi_ip {
```

Comment:
Nit: could do `if let (Some(core_ip_location), Some(efi_ip)) = (core_ip, efi_ip) {`. Not sure if that's particularly more or less clear than if-let-chains.